### PR TITLE
Issue #1728: Introduce `layout_suppress()` as a clean method to suppress layouts.

### LINKED ISSUE #1728: Please provide a clean method to suppress layout.
I need a method to show pages with "content only".

A hack based on `$_SERVER['HTTP_ACCEPT']` has been suggested in https://github.com/backdrop/backdrop-issues/issues/1483#issuecomment-169173412 but it won't work in 1.3.4 with https://github.com/backdrop/backdrop/pull/1290

This is currently used in my port of [References dialog](https://github.com/backdrop-contrib/references_dialog)
Do I have to commit another hack based on a fake `$_SERVER['HTTP_X_REQUESTED_WITH']` ?

A `function layout_suppress()`, similar to `admin_bar_suppress()` would be convenient.
(just to request `layout_route_handler()` tu use `menu_default_route_handler()`)

Note: the drupal version uses a query `?render=references-dialog` to fire a page template...

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/1728

Based off of @gifad's PR here: https://github.com/backdrop/backdrop/pull/1300

I have:

- resolved conflicts with recent 1.x
- merged the new conditional that returns `menu_default_route_handler($router_item)`, with the one directly above it (and adapted the docbloc accordingly).

### CODE REVIEW

File: core/modules/layout/layout.module [hunk 1]
```diff
@@ -536,6 +536,27 @@ function layout_page_access() {
...
+ * useful for modules that implement popup pages or other special pages, where
+ * the default layout would be distracting, or just wasted space.
+ *
+ * @param $suppress
+ *   If set to TRUE, and called before menu_execute_active_handler(), the layout
+ *   will default to menu_default_route_handler().
+ *   The suppression state is returned.
+ */
+function layout_suppress($suppress = NULL) {
+  static $suppression_state = FALSE;
```
Comment (indigoxela) [inline core/modules/layout/layout.module]: Maybe someone can tell me: why "static"?

From above comment my understanding is that this runs *once* in hook_init(). Can there ever be a situation, where this code runs more than once per page call? IMO only then the static would make sense. But maybe I'm missing something.

Comment (herbdool) [reply core/modules/layout/layout.module]: I think so it behaves like [backdrop_page_is_cacheable](https://docs.backdropcms.org/api/backdrop/core%21includes%21bootstrap.inc/function/backdrop_page_is_cacheable/1) as mentioned here https://github.com/backdrop/backdrop-issues/issues/1728#issuecomment-200631379

Comment (indigoxela) [reply core/modules/layout/layout.module]: I've no permission to do so, but consider this comment/question as outdated/resoved. :+1:

File: core/modules/layout/layout.module [hunk 2]
```diff
@@ -577,6 +577,28 @@ function layout_page_access() {
...
+ * This function should be called from within another module's page callback
+ * when the layout should not be displayed, that is, content only. This can be
+ * useful for modules that implement popup pages or other special pages where
+ * the default layout would be distracting.
+ *
+ * @param bool $suppress
+ *   If set to TRUE, the layout will be supressed.
+ * @return bool
+ *   TRUE if the layout can be suppressed, FALSE otherwise.
+ */
```
Comment (quicksketch) [inline core/modules/layout/layout.module:594]:
```suggestion
 *
 * @since 1.20.1 Function added.
 */
```

**Comment by quicksketch** [comment]
Merged into 1.x and 1.20.x. Thanks! I also closed https://github.com/backdrop/backdrop/pull/1300.

**Comment by quicksketch** [comment]
Oops, sorry this was a 1.21.x issue only! I reverted the commit to 1.20.x and this will be in 1.21.0 only.

**Pull request merged**